nimterop/grammar.nim

```diff
@@ -492,12 +499,9 @@ proc initGrammar(): Grammar =
 
         if fnname.nBl and nimState.identifiers.addNewIdentifer(fnname):
           let ftyp = nimState.data[0].val.getIdentifier(nskType, fnname)
-
-          if fptr == "ptr " or ftyp != "object":
-            nimState.procStr &= &"proc {fnname}*({pout}): {getPtrType(fptr&ftyp)} {{.importc: \"{fname}\", header: {nimState.currentHeader}.}}\n"
-          else:
-            nimState.procStr &= &"proc {fnname}*({pout}) {{.importc: \"{fname}\", header: {nimState.currentHeader}.}}\n"
-
+          var returnTypeStr = ""
+          if fptr == "ptr " or ftyp != "object": returnTypeStr = &": {getPtrType(fptr&ftyp)}"
+          nimState.procStr &= &"proc {fnname}*({pout}){returnTypeStr} {{.{genImportC(fname, fnname)}, header: {nimState.currentHeader}.}}\n"
```

Comment:
Let's leave this `if ptr` pattern as is since it is in multiple places. Or simplify all of them in the same style.